File: C#/프로그래머스/0/181906. 접두사인지 확인하기/접두사인지 확인하기.cs
using System;
using System.Linq;
using System.Collections.Generic;

public class Solution {
    public int solution(string my_string, string is_prefix)
    {
        char[] array = my_string.ToCharArray();
        List<string> list = new List<string>();
        
        for(int x=0;x<my_string.Length;x++)
        {
            string word = "";
            for(int y=0;y<=x;y++)
            {
                word += array[y];
            }
            if(word != "")
            {
                list.Add(word);
            }
        }
        if(list.Contains(is_prefix))
        {
            return 1;
        }
        else
        {
            return 0;
        }
    }
}

File: C/프로그래머스/0/181906. 접두사인지 확인하기/접두사인지 확인하기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

// 파라미터로 주어지는 문자열은 const로 주어집니다. 변경하려면 문자열을 복사해서 사용하세요.
int solution(const char* my_string, const char* is_prefix) 
{
    //is_prefix의 (문자열+NULL)의 길이 알아내기
    int count = 0;
    for (int x = 0;;x++)
    {
        if (is_prefix[x] == NULL)
        {
            count = x;
            break; 
        }
    }
    
    
    bool isCorrect = true;
    
    for(int x=0;x<count;x++) //NULL문자 제외
    {
        if((my_string[x] == NULL && is_prefix[x] != NULL)) //비교할 문자열이 더 긴 경우 제외.  
        {
            isCorrect = false;
            break;
        }  
        if(my_string[x] != is_prefix[x])
        {
            isCorrect = false;
            break;
        }
    }
    
    return isCorrect;
}
